src/almanack/git_parser.py

```diff
@@ -49,19 +56,22 @@ def get_commit_contents(repository_path: str, commit_id: str):
     return contents
 
 
-def collect_all_commit_logs(repositories: str):
+def calculate_loc_changes(repo_path: pathlib.Path, source: str, target: str) -> int:
     """
-    Gather commit logs for each repository in the given collection.
+    Finds the total number of code lines changed between the source or target commits.
 
     Args:
-        repository_path (str): The path to the Git repository.
-
+        repo_path (pathlib.Path): The path to the git repository.
+        source (str): The source commit hash.
+        target (str): The target commit hash.
     Returns:
-        dict: A dictionary mapping repository names to their commit logs.
-            Example: {'repository_name': {'commit_id': {'message': 'Commit message', 'timestamp': 1234567890}}}
+        dict: A dictionary where the key is the filename, and the value is the lines changed (added and removed)
+            Example: {'filename': 'change_value'}
     """
-    all_logs = {}
-    for repo_name, repo_path in repositories.items():
-        # Retrieve commit logs for each repository and store them in the dictionary
-        all_logs[repo_name] = get_commit_logs(repo_path)
-    return all_logs
+    repo = git.Repo(repo_path)
+    # diff(--numstat) provides the number of added and removed lines for each file
+    diff = repo.git.diff(source, target, "--numstat")
+    return {
+        filename: abs(int(removed) + int(added))  # Calculate change
+        for added, removed, filename in (line.split() for line in diff.splitlines())
```

Comment:
Given how you gather data from this (data within a string) would it make sense to expand the test data to ensure nothing unusual happens when you have more than one file in a repository? Mostly I wonder, what would happen if there were two files (perhaps with one in a subdir and the other not)?